File: src/main/java/com/fishercoder/solutions/_1295.java
package com.fishercoder.solutions;

import java.util.Arrays;

public class _1295 {
    public static class Solution1 {
        public int findNumbers(int[] nums) {
            int count = 0;
            for (int num : nums) {
                if (String.valueOf(num).length() % 2 == 0) {
                    count++;
                }
            }
            return count;
        }
    }

    public static class Solution2 {
        public int findNumbers(int[] nums) {
            return (int) Arrays.stream(nums).filter(num -> String.valueOf(num).length() % 2 == 0).count();
        }
    }
}

File: javascript/_1295.js
/**
 * @param {number[]} nums
 * @return {number}
 */

var findNumbers = function (nums) {
  let result = 0
  for (let i = 0; i < nums.length; i++) {
    const num = String(nums[i])
    if (num.length % 2 === 0) result++
  }
  return result
}

console.log(findNumbers([12, 345, 2, 6, 7896]))
